1/tests/unit/ListItem.spec.js

```diff
@@ -0,0 +1,76 @@
+import { shallowMount } from '@vue/test-utils';
+import ListItem from '@/components/ListItem.vue';
+import { mount } from '@vue/test-utils';
+
+describe('ListItem.vue', () => {
+	it('render props.message when passed', () => {
+		const message = 'new todo';
+		const wrapper = shallowMount(ListItem, {
+			propsData: { message }
+		});
+		expect(wrapper.text()).toMatch(message);
+	});
+});
+
+describe('ListItem', () => {
+	describe('given a `todo`', () => {
+		const wrapper = mount(ListItem, {
+			name: 'item',
+			propsData: {
+				id: 1,
+				message: 'Foo'
+			}
+		});
+
+		
+		it('renders `item.message`', () => {
+			expect(wrapper.vm.message).toBe('Foo');
+		});
+		it("doesn't show input field", () => {
+			expect(wrapper.vm.onEdit).toBeFalsy();
+		});
+		it('has the expected html structure', () => {
+			expect(wrapper.element).toMatchSnapshot();
+		});
+
+		describe('Click on `Edit` button', () => {
+			it('shows input field', () => {
+				const aArray = wrapper.findAll('a');
+				const Edit = aArray.at(0);
+				Edit.trigger('click');
```

Comment:
Why not `wrapper.find('a').trigger('click')` ?

According to the docs it's choosing the first element if there are many: https://vue-test-utils.vuejs.org/api/wrapper/#find